account_commission_sales_fee/models/commission_settlement.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2022 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class CommissionSettlement(models.Model):
+    _inherit = "commission.settlement"
+
+    def _prepare_sales_fee_product(self):
+        sales_fee_product = self.env["product.product"].search(
+            [("sales_fee", "=", True)], limit=1
+        )
+        if not sales_fee_product:
+            raise UserError(
+                _(
+                    "You don't have the sales fee product. Please create sales fee product."
+                )
+            )
+        return {
+            "product_id": sales_fee_product.id,
+            "quantity": len(self.line_ids),
+            "price_unit": sales_fee_product.standard_price * -1,
+        }
```

Comment:
Can we add an indicator (say, `calculate_sales_fee`) in the commission model, and do this computation only when the indicator is True to avoid forcing this calculation all the time?  In that case, actually, we can add `sales_fee_product_id` to the commission model instead of `sales_fee` to product.template.  What do you think?

Reply:
> Instead of adding `calculate_sales_fee` and `sales_fee_product_id` in commission model, we should add in wizard form of make invoice and we only need to choose `sales_fee_product_id` when `calculate_sales_fee` is True.

Disagreed.  We'd like to control the behavior based on the commission type (`commission` record).  Adding the ability to adjust whether or not to calculate the sales fee in the wizard (where default settings are proposed from the relevant commission record) could be something nice to have, but we do not need that at  least for now.

> And I prefer only `service` type product can be chosen in `sales_fee_product_id`.

Agreed.